src/datasource.ts

```diff
@@ -126,33 +126,19 @@ export class GenericDatasource {
         // remove placeholder targets
         return target.target !== 'select metric';
       })
-      .map((target: any) => {
-        const data = isUndefined(target.data) || target.data.trim() === '' ? null : JSON.parse(target.data);
-
-        if (data !== null) {
-          Object.keys(data).forEach(key => {
-            const value = data[key];
-            if (typeof value !== 'string') {
-              return;
+      .map((target) => {
+        let data = null;
+
+        if (!isUndefined(target.data) && target.data.trim() !== '') {
+          data = JSON.parse(target.data, (key, value) => {
+            if (typeof value === 'string') {
+              return value.replace(
+                this.templateSrv.regex,
+                match => this.cleanMatch(match, options),
+            );
             }
```

Comment:
There's some issue with CI, I'll try to reconfigure it but this should not pass lint then.

Reply:
@vankop I know why CI is not being triggered. Can you squash commits and rebase on top of master? There were some changes in CI config that are not present in this branch.